Set the attenuation of a specific DDS

BECATRUE opened the pull request:
This closes #81.

For a request, refer to the following command.
```shell
$ curl -X POST 'http://127.0.0.1:8000/dds/att/?device=urukul0&channel=1&value=3.5'
```

BECATRUE committed:
Define `set_dds_attenuation()`

See also: #81

BECATRUE committed:
Check if DDS channel is valid

See also: #81

BECATRUE committed:
Write the code to set att

See also: #81

BECATRUE committed:
Submit the corresponding experiment

See also: #81

kangz12345 commented on main.py:
```diff
@@ -634,6 +634,46 @@ def run(self):
 
 
+@app.post("/dds/att/")
+async def set_dds_attenuation(device: str, channel: int, value: float):
```
Please annotate the return type. Also, note that it may return `None`. Or you could give for example -1 for the return value.

BECATRUE replied:
As your suggestion, it looks better to return -1 instead of returing nothing.

kangz12345 commented on main.py:
```diff
@@ -634,6 +634,46 @@ def run(self):
+    """
+    if device not in configs["dds_devices"] or channel not in configs["dds_devices"][device]:
+        logger.exception("The DDS device %s CH %d is not defined in config.json.", device, channel)
+        return
```
Since this function returns something in usual cases, please explicitly write `return None`.

See also: [PEP8](https://peps.python.org/pep-0008/#programming-recommendations)
> Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, any return statements where no value is returned should explicitly state this as `return None`, and an explicit return statement should be present at the end of the function (if reachable)

BECATRUE replied:
As you suggested above, I returned -1 instead of `None`.

BECATRUE committed:
Annotate return type

See also: #81

kangz12345 commented on main.py:
```diff
@@ -634,6 +634,50 @@ def run(self):
+        If there is an error, it returns -1.
+    """
+    if device not in configs["dds_devices"] or channel not in configs["dds_devices"][device]:
+        logger.exception("The DDS device %s CH %d is not defined in config.json.", device, channel)
```
Please use `error()` instead of `exception()`.

BECATRUE committed:
Use `error()` instead of `exception()`

See also: #81

kangz12345 reviewed (dismissed):
LGTM

BECATRUE committed:
Relocate two functions

See also: #81

BECATRUE commented:
To resolve the conflicts, I rebased the latest develop branch on this branch and pushed it forcely.
Please recheck it!

kangz12345 commented:
I think merge commit (conflict resolving) is a better way to resolve conflicts. Force-push should be avoided if possible.
Reviewers need to read the whole code again and it is hard to check if something goes wrong!

kangz12345 commented:
BTW, where was the conflict?

BECATRUE commented:
> BTW, where was the conflict?

After merging #83, the `set_dds_switch()` function appeared just after the `set_dds_profile()`.
In this PR, however, the `set_dds_attenuation()` is also implemented just after the `set_dds_profile()` and `set_dds_attenuation()` is similar to `set_dds_switch()`, hence I thought GitHub judged that I changed `set_dds_switch()` implemented in #83.

kangz12345 commented:
Okay! In that case, you can resolve it in GitHub as well.

See also: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github
